src/app/shared/filesystem/DirectoryTree.ts

```diff
@@ -65,10 +64,36 @@ export class DirectoryTree {
    * @memberof DirectoryTree
    */
   private cleanFilesystem(files: LearningObject.Material.File[]) {
-    const fileIds = files.map(file => file.id);
+    /**
+     * Checks if file is in array by matching against id and path
+     *
+     * @param {{
+     *       cachedFileData: { id: string; path: string };
+     *       file: LearningObject.Material.File
+     *     }} params
```

Comment:
I think that this is auto-generated JSDoc from a tool. I'd vote that we either remove it, or take some time to write doc that explains these properties. Currently this doesn't do anything that TypeScript doesn't already do.

I've seen it a lot throughout our code and I think it just clutters things up.